$OpenBSD$

System.Random.genByteString removed in random >= 1.2.1. Provide local
implementation guarded by MIN_VERSION_random(1,2,1) CPP check.

--- a/src/Simplex/Messaging/Server/MsgStore/Journal.hs
+++ b/src/Simplex/Messaging/Server/MsgStore/Journal.hs
@@ -92,7 +92,13 @@
 import System.FilePath (takeFileName, (</>))
 import System.IO (BufferMode (..), Handle, IOMode (..), SeekMode (..))
 import qualified System.IO as IO
+#if MIN_VERSION_random(1,2,1)
+import System.Random (StdGen, newStdGen, randomR)
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+#else
 import System.Random (StdGen, genByteString, newStdGen)
+#endif
 
 data JournalMsgStore s = JournalMsgStore
   { config :: JournalStoreConfig s,
@@ -819,6 +825,14 @@
 newJournalId :: TVar StdGen -> IO ByteString
 newJournalId g = strEncode <$> atomically (stateTVar g $ genByteString 12)
 
+#if MIN_VERSION_random(1,2,1)
+genByteString :: Int -> StdGen -> (ByteString, StdGen)
+genByteString n g = go n g []
+  where
+    go 0 g' acc = (BS.pack (reverse acc), g')
+    go i g' acc = let (w, g'') = randomR (0, 255 :: Int) g' in go (i - 1) g'' (fromIntegral w : acc)
+#endif
+
 openJournals :: JournalMsgStore s -> FilePath -> MsgQueueState -> Handle -> IO (MsgQueueState, Handle, Maybe Handle)
 openJournals ms dir st@MsgQueueState {readState = rs, writeState = ws} sh = do
   let rjId = journalId rs
